### PR TITLE
Refactor pauser defaults. De-deprecate

### PR DESCRIPTION
**Pull request by JerryShea**
See also https://github.com/ChronicleEnterprise/Chronicle-Queue-Enterprise/pull/391 & https://github.com/ChronicleEnterprise/Chronicle-Map-Enterprise/pull/112

Signed-off-by: Jerry Shea <jerry.shea@chronicle.software>

### CODE REVIEW

File: src/main/java/net/openhft/chronicle/network/cluster/ClusterContext.java [hunk 1]
```diff
@@ -133,7 +133,7 @@ public void accept(HostDetails hd) {
 
         if (hd.connectUri() == null)
             return;
-        acceptorLoop = new BlockingEventLoop(eventLoop(), clusterNamePrefix() + "acceptor-" + localIdentifier);
+        acceptorLoop = new BlockingEventLoop(eventLoop(), clusterNamePrefix() + "acceptor-" + localIdentifier, Pauser.balanced());
```
Comment (JerryShea) [inline src/main/java/net/openhft/chronicle/network/cluster/ClusterContext.java:136]: don't call deprecated constructor

**Comment by JerryShea** [comment]
[build all branch](https://teamcity.chronicle.software/project/Chronicle_BuildAll?branch=feature%2Fimprove-pauser-defaults&mode=builds) green

File: src/main/java/net/openhft/chronicle/network/cluster/ClusterContext.java [hunk 2]
```diff
@@ -226,7 +226,10 @@ public void cluster(Cluster<T, C> cluster) {
         this.cluster = cluster;
     }
 
-    protected abstract void defaults();
+    protected void defaults() {
```
Comment (alamar) [inline src/main/java/net/openhft/chronicle/network/cluster/ClusterContext.java:229]: I wonder if that breaks existing bytecode

**Review by alamar — APPROVED** [review]

File: src/main/java/net/openhft/chronicle/network/cluster/ClusterContext.java (hunk 2 above)
Comment (JerryShea) [reply src/main/java/net/openhft/chronicle/network/cluster/ClusterContext.java:229]: It does not :)
I have tried running QE `mvn surefire:test` before and after replacing the jar and all good

Comment (nicktindall) [reply src/main/java/net/openhft/chronicle/network/cluster/ClusterContext.java:229]: We will probably need to add calls to `super.defaults()` in the places `defaults` was previously implemented, now overridden? I know there's two tests that do it and potentially more in other repos.

**Review by nicktindall — APPROVED** [review]
Pending a review of places where `defaults` is overridden

File: src/main/java/net/openhft/chronicle/network/cluster/ClusterContext.java (hunk 2 above)
Comment (JerryShea) [reply src/main/java/net/openhft/chronicle/network/cluster/ClusterContext.java:229]: DG, Map-Enterprise and Queue-Enterprise override it and all are now working as expected - see https://github.com/ChronicleEnterprise/Chronicle-Queue-Enterprise/pull/391 & https://github.com/ChronicleEnterprise/Chronicle-Map-Enterprise/pull/112

All the tests are green - [build all branch](https://teamcity.chronicle.software/project/Chronicle_BuildAll?branch=feature%2Fimprove-pauser-defaults&mode=builds) - so I think we are good.

Not appropriate to call super.defaults in all overrides as then difficult to override default value of e.g. pauserSupplier

File: src/main/java/net/openhft/chronicle/network/cluster/ClusterContext.java [hunk 3]
```diff
@@ -170,7 +170,7 @@ protected synchronized EventLoop synchronizedEventLoop() {
             return el;
 
         return this.eventLoop = EventGroupBuilder.builder()
-                .withPauser(pauserSupplier.get())
+                .withPauser(pauserSupplier != null ? pauserSupplier.get() : null)
```
Comment (JerryShea) [inline src/main/java/net/openhft/chronicle/network/cluster/ClusterContext.java:173]: make tolerant of null - use the event group builder's default

File: src/main/java/net/openhft/chronicle/network/cluster/ClusterContext.java [hunk 4]
```diff
@@ -226,7 +226,10 @@ public void cluster(Cluster<T, C> cluster) {
         this.cluster = cluster;
     }
 
-    protected abstract void defaults();
+    protected void defaults() {
+        pauserSupplier = DEFAULT_PAUSER_MODE;
```
Comment (JerryShea) [inline src/main/java/net/openhft/chronicle/network/cluster/ClusterContext.java:230]: these are the same defaults that were in field initialiser - now just called here, where they can be easily overridden

**Pull request merged**